appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/poa/manager/PoAManager.java

```diff
@@ -203,11 +203,12 @@ private void sendProof() {
 
         Log.e(TAG, "Proof " + (proofsSent + 1) + " skipped! Came from background!");
       } else {
-        // Connection to service may already been done, but we still need to make sure that it is
-        // connected. In case no connection is not yet done, the message is stored to be sent as soon as
-        // the connection is done.
-        poaConnector.connectToService(appContext);
-        // send proof
+        if (!firstProof) {
```

Comment:
Don't create a variable to control the first proof sent when we already have a variable that controls the number of proofs sent, in this case we already have the 'proofsSent' variable to control the number of proofs sent.

Reply:
Maybe use the logic to postpone the proof to be sent on the initiateProofSending method instead of doing it in the sendProof method